lib/pbench/server/unpack_tarballs.py

```diff
@@ -42,18 +51,60 @@ def unpack_tarball(self) -> (int, int):
                 tb = Path(tarball).resolve(strict=True)
             except FileNotFoundError:
                 self.logger.error(
-                    "Tarball link, '{}', does not resolve to a real location", tarball
+                    "{}: Tarball link, '{}', does not resolve to a real location",
+                    self.config.TS,
+                    tarball,
                 )
                 continue
 
-            tar_md5 = get_tarball_md5(tb)
+            try:
+                tar_md5 = get_tarball_md5(tb)
+            except FileNotFoundError as exc:
+                self.logger.error(
+                    "{}: Getting md5 value of tarball '{}' failed: {}",
+                    self.config.TS,
+                    tb.name,
+                    exc,
+                )
+                continue
 
             try:
                 file_tree.unpack(tar_md5)
             except Exception as exc:
-                self.logger.error("Unpacking of tarball {} failed: {}", tb, exc)
+                self.logger.error(
+                    "{}: Unpacking of tarball {} failed: {}",
+                    self.config.TS,
+                    tb.name,
+                    exc,
+                )
+                continue
+
+            linkdestlist = self.config.get("pbench-server", "unpacked-states").split(
+                ","
+            )
+            controller_name = tb.parent.parent.name
+            dest = self.config.ARCHIVE / controller_name
+
+            for state in linkdestlist:
+                try:
+                    os.symlink((dest / tb.name), (dest / state / tb.name))
+                except Exception as exc:
+                    self.logger.error(
+                        "{}: Error in creation of symlink. '{}'", self.config.TS, exc
```

Comment:
I don't think it's helpful to have the single-quotes around the exception message:  the message already begins with a single-quote, and so we end up with a double single-quote in the log message, and it looks funny.  (Of course, if you fix the code here, you'll need to make the companion fix to the unit test.)